docs/source/user-guide/tabular/feature_importance.rst

```diff
@@ -76,14 +77,15 @@ function, available in all checks and suites.
 
 
 What if the Feature Importance Calculation Takes Too Long?
-==========================================================
+=========================================================
 
-Feature importance is a complex calculation which can take a lot of time, depending on the number of features and
+Permutation feature importance is a complex calculation which can take a lot of time, depending on the number of features and
 samples in your data.
-However, besides for cetrain checks, deepchecks does not require feature importance.
+However, except for certain checks, deepchecks does not require feature importance.
 Therefore, if you want deepchecks to skip the calculation of feature importance, you can use the
-``feature_importance_timeout`` parameter in the ``run`` function, available in all checks and suites. If
-deepchecks projects the feature importance calculation to take more than this parameter, the process will be skipped.
+``feature_importance_timeout`` parameter in the ``run`` function, available in all checks and suites.
+Before running the permutation feature importance, deepchecks predicts the calculation time. If the predicted time
+is bigger than ``feature_importance_timeout``, the process will be skipped.
 Configuring this parameter to 0 will ensure the calculation is always skipped.
```

Comment:
Are we sure about this behaviour? I would have thought setting it to 0 mean run for as long as it takes

Reply:
yes, sure :)